### PR TITLE
Implement PMS.complete method for WriteHdfsP to commit the job

### PR DESCRIPTION
**Pull request by gurbuzali**
fixes https://github.com/hazelcast/hazelcast-jet/issues/565

### CODE REVIEW

**Comment by cangencer** [comment]
can we add a test for this?

**Comment by gurbuzali** [comment]
I did write a test for that but it is not failing even without the fix. I guess it is because our tests are using `LocalFileSystem` not `DistributedFileSystem`.

File: hazelcast-jet-hadoop/src/main/java/com/hazelcast/jet/impl/connector/hadoop/WriteHdfsP.java [hunk 1]
```diff
@@ -161,18 +169,15 @@ public void init(@Nonnull Context context) {
 
         @Override
         public void complete(Throwable error) {
-            if (commitJob) {
-                uncheckRun(() -> outputCommitter.commitJob(jobContext));
+            if (processorList != null) {
+                processorList.forEach(Processor::complete);
```
Comment (cangencer) [inline hazelcast-jet-hadoop/src/main/java/com/hazelcast/jet/impl/connector/hadoop/WriteHdfsP.java]: wouldn't this cause Processor.complete() to be called twice? Once after DAG finishes, and once on PS.complete()?

Comment (gurbuzali) [reply hazelcast-jet-hadoop/src/main/java/com/hazelcast/jet/impl/connector/hadoop/WriteHdfsP.java]: Yes, I've missed to change it to a private `close` method, fixed now.

**Pull request merged**